### PR TITLE
farm#422 title declared in agenda.xsd

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for zerocracy/farm#422, added the title in the xsd + modified ``.xml`` tests and ``agenda.xsl``

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#331](https://github.com/zerocracy/datum/issues/331) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#331](https://github.com/zerocracy/datum/issues/331) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/760849629). The budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: xml/pmo/agenda/-missing-title.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<agenda version="0.25" updated="2016-12-29T09:03:21.684Z">
+    <order job="gh:test/test#1">
```
Comment (carlosmiranda) [inline xml/pmo/agenda/-missing-title.xml]: @amihaiemil Qulice is complaining about the formatting of this XML.

**Comment by amihaiemil** [comment]
@carlosmiranda I'm not sure what's wrong -- the fact that the line endings are ``\n``?

**Comment by amihaiemil** [comment]
@carlosmiranda anyway, I got another question, how do we continue with the puzzle tree? Yegor suggested how we should continue [here](https://github.com/zerocracy/farm/pull/784#issuecomment-372447596) but I am not sure where to leave the puzzle for that... should I make another PR in /farm? Seems useless, it's cheaper and for the project if someone opens an Issue instead.

File: xml/pmo/agenda/-missing-title.xml (hunk 1 above)
Comment (carlosmiranda) [reply xml/pmo/agenda/-missing-title.xml]: @amihaiemil could be the indentation. Our other XMLs use 2 space indentation.

**Comment by carlosmiranda** [comment]
@amihaiemil leave it in farm, if that's where the next step is relevant.

**Comment by amihaiemil** [comment]
@carlosmiranda qulice fixed now, thanks

**Comment by carlosmiranda** [comment]
@yegor256 looks good to merge here

**Comment by yegor256** [comment]
@amihaiemil why it's an attribute? XML attributes are used for meta-information, while XML elements -- for data. Make it `<title>` instead

**Comment by amihaiemil** [comment]
@yegor256 done; can you have a look?

**Comment by amihaiemil** [comment]
@yegor256 Can we also merge this?

**Comment by yegor256** [comment]
@amihaiemil the changes you are introducing will crash all existing agenda's XML files, since they don't have `<title>` elements. They will become XSD non-conformant and the Farm will stop to work. In order to avoid that we have so-called "upgrades", which are XSL transformations that are modifying existing XML documents if their versions are too low. See: af69f8f

**Comment by amihaiemil** [comment]
@yegor256 I see, I was also wondering about that... I'll have a look today

**Comment by amihaiemil** [comment]
@yegor256 I added the upgrade and reversed the ``missing-title`` test, now we expect it to pass since the title is added by the ``upgrade``.

File: xsd/pm/scope/types.xsd [hunk 2]
```diff
@@ -21,6 +21,10 @@ SOFTWARE.
       <xs:pattern value="none|gh:[A-Z0-9a-z.\-#/]{3,}"/>
     </xs:restriction>
   </xs:simpleType>
+  <xs:simpleType name="jobTitle">
+    <xs:restriction base="xs:string">
+    </xs:restriction>
```
Comment (yegor256) [inline xsd/pm/scope/types.xsd]: @amihaiemil I think we should add some regexp here, at lease `.{3,}`

**Comment by amihaiemil** [comment]
@yegor256 I was thinking about restirctions too, but then we would have to comply 1:1 with Github/Bitbucket/Jira etc. For instance, Github allows titles with only 1 character. Still think we should add our own rules?

**Comment by yegor256** [comment]
@amihaiemil yes, let's go with "at least one character" rule

**Comment by amihaiemil** [comment]
@yegor256 done

**Comment by yegor256** [comment]
@rultor merge

**Comment by amihaiemil** [comment]
@yegor256 I've synced the branch with master, can we try to merge again?

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 13days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 13days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by carlosmiranda** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/datum/issues/331#issuecomment-378457079))

@carlosmiranda This is what I know about this job, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#331](https://github.com/zerocracy/datum/issues/331) is [in scope](http://www.zerocracy.com/policy.html#14) for 18days
  * The role is `REV`
  * The job is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) for [18days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will get 25% of their hourly rate for this job
  * Some equity will be [vested](http://www.zerocracy.com/policy.html#37) on completion
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @amihaiemil[/z](https://www.0crat.com/u/amihaiemil): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Fdatum%23331%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by carlosmiranda** [comment]
@yegor256 can we take this out of scope, `0crat` seems to be stuck on it

**Comment by yegor256** [comment]
@0crat out

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/zerocracy/datum/issues/331#issuecomment-379148647))

@yegor256 The job [#331](https://github.com/zerocracy/datum/issues/331) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@carlosmiranda According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one cosmetic issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>All commit messages have links to issues.

Please confirm that you'll try to follow this rule next time.

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed

**Comment by amihaiemil** [comment]
@ypshenychka I confirm :)

**Comment by ypshenychka** [comment]
@amihaiemil Thanks.

**Comment by ypshenychka** [comment]
@carlosmiranda Thank you.

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**